Update AutoML-NLP.md (#462)
microsoft/FLAML: website/docs/Examples/AutoML-NLP.md

### A simple sequence classification example

> To use the sequence classification module in FLAML, simply load your data, convert it to the pandas format, and call automl.fit:

Reviewer: ```suggestion
To perform the sequence classification task in FLAML, simply load your data, convert it to the pandas format, and use `AutoML.fit()`:
```

```python
from flaml import AutoML
from datasets import load_dataset

> train_dataset = load_dataset("glue", "mrpc", split="train").to_pandas()  

Reviewer: ```suggestion
train_dataset = load_dataset("glue", "mrpc", split="train").to_pandas()
```

### A simple sequence classification example

An example of sequence classification data is the Stanford Sentiment Treebank data (SST) data:

sentence  label 
hide new secretions from the parental units       0 

> contains no wit , only labored gags       0 

Reviewer: include one positive example.
use markdown to format it so that it's clear to see it's a table.

dev_dataset = (
    load_dataset("glue", "stsb", split="train").to_pandas()
)

custom_sent_keys = ["sentence1", "sentence2"]   # specify the column names of the input sentences
label_key = "label"                             # specify the column name of the label

X_train = train_dataset[custom_sent_keys]
y_train = train_dataset[label_key]
X_val = dev_dataset[custom_sent_keys]
y_val = dev_dataset[label_key]

automl = AutoML()
automl_settings = {

>     "gpu_per_trial": 0,   

Reviewer: ```suggestion
    "gpu_per_trial": 0,
```

    X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val, **automl_settings
)
```

### A simple token classification example

The token classification can be used for tasks such as named entity recognition and part-of-speech tagging. An example of using FLAML for NER is:

```python
from flaml import AutoML
from datasets import load_dataset

train_dataset = (
    load_dataset("conll2003", split="train").to_pandas()
)
dev_dataset = (
    load_dataset("conll2003", split="validation").to_pandas()
)

custom_sent_keys = ["tokens"]                   # specify the column names of the input data
label_key = "ner_tags"                          # specify the column name of the output token tags

X_train = train_dataset[custom_sent_keys]
y_train = train_dataset[label_key]

X_val = dev_dataset[custom_sent_keys]
y_val = dev_dataset[label_key]

automl = AutoML()
automl_settings = {
    "gpu_per_trial": 0,

>     "max_iter": 2,
>     "time_budget": 5,                       # specify the time budget 

Reviewer: ```suggestion
    "time_budget": 50,                       # specify the time budget 
```

        "task": "token-classification",         # specify the task to token classification
        "metric": "seqeval",                    # specify the metric

>     }

Reviewer: fix indent

automl_settings["custom_hpo_args"] = {
        "model_path": "bert-base-uncased",
        "output_dir": "test/data/output/",
        "ckpt_per_epoch": 1,
        "fp16": False,

>     }

Reviewer: fix indentation

On stage, a woman takes a seat at the piano. She
a) sits on a bench as her sister plays with the doll.
b) smiles with someone as the music plays.
c) is in the crowd, watching the dancers.
d) nervously sets her fingers on the keys

An example of running FLAML for the multiple choice task:

```python
from flaml import AutoML
from datasets import load_dataset

train_dataset = (
    load_dataset("swag", split="train").to_pandas()
)
dev_dataset = (
    load_dataset("swag", split="validation").to_pandas()
)

custom_sent_keys = [                            # specify the column names of the first sentence and choices
    "sent1",
    "sent2",
    "ending0",
    "ending1",
    "ending2",
    "ending3",
    "gold-source",
    "video-id",
    "startphrase",
    "fold-ind",
]
label_key = "label"                         # specify the column name of the output token tags

X_train = train_dataset[custom_sent_keys]
y_train = train_dataset[label_key]

X_val = dev_dataset[custom_sent_keys]
y_val = dev_dataset[label_key]

automl = AutoML()
automl_settings = {
    "gpu_per_trial": 0,

>     "max_iter": 2,
>     "time_budget": 5,                      # specify the time budget 

Reviewer: ```suggestion
    "time_budget": 50,                      # specify the time budget 
```

        "task": "multichoice-classification",  # specify the task to multiple choice
        "metric": "accuracy",                  # specify the evaluation metric
        "log_file_name": "seqclass.log",

>     }

Reviewer: fix indentation

automl_settings["custom_hpo_args"] = {
        "model_path": "google/electra-small-discriminator",
        "output_dir": "test/data/output/",
        "ckpt_per_epoch": 1,
        "fp16": False,

>     }

Reviewer: fix indentation
